cupyx/array_api/_typing.py

```diff
@@ -0,0 +1,47 @@
+"""
+This file defines the types for type annotations.
+
+These names aren't part of the module namespace, but they are used in the
+annotations in the function signatures. The functions in the module are only
+valid for inputs that match the given type annotations.
+"""
+
+from cupy.cuda import Device as _Device
+
+
+__all__ = [
+    "Array",
+    "Device",
+    "Dtype",
+    "SupportsDLPack",
+    "SupportsBufferProtocol",
+    "PyCapsule",
+]
+
+from typing import Any, Sequence, Type, Union
+
+from . import (
+    Array,
```

Comment:
Hi @asmeurer I am confused by these two lines. `Array` does not live in either `numpy.arrray_api` or `numpy.arrray_api._typing`, does it? So why does this import statement work?

Reply:
It looks like it doesn't work actually. I guess this file is never explicitly imported anywhere. 

```
>>> import numpy.array_api._typing
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "./numpy/array_api/_typing.py", line 20, in <module>
    from . import (
ImportError: cannot import name 'Array' from 'numpy.array_api' (./numpy/array_api/__init__.py)
```

Reply:
hmmm... @asmeurer then it looks like the NumPy code would not work with any static type analyzer?
```python
>>> import typing
>>> typing.TYPE_CHECKING
False
>>> typing.TYPE_CHECKING = True
>>> from cupyx import array_api
<stdin>:1: UserWarning: The numpy.array_api submodule is still experimental. See NEP 47.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/leof/dev/cupy_cuda114/cupyx/array_api/__init__.py", line 133, in <module>
    from ._creation_functions import (
  File "/home/leof/dev/cupy_cuda114/cupyx/array_api/_creation_functions.py", line 7, in <module>
    from ._typing import (
  File "/home/leof/dev/cupy_cuda114/cupyx/array_api/_typing.py", line 23, in <module>
    from . import (
ImportError: cannot import name 'Array' from partially initialized module 'cupyx.array_api' (most likely due to a circular import) (/home/leof/dev/cupy_cuda114/cupyx/array_api/__init__.py)
```